saleor/graphql/giftcard/resolvers.py

```diff
@@ -0,0 +1,12 @@
+import graphene
+
+from .types import GiftCard
+
+
+def resolve_gift_card(info, gift_card_id):
+    """Return gift card only for user assigned to it or proper staff user."""
+    user = info.context.user
+    gift_card = graphene.Node.get_node_from_global_id(info, gift_card_id, GiftCard)
+    if user.has_perm("giftcard.manage_gift_card") or gift_card.creator == user:
```

Comment:
I would think we should check first if the user has the permission or not.
- If they have: just retrieve it.
- If they don't have it: `get_node_optimized(qs, {"pk": gift_card_id, "creator": user})`

So we don't get a db query when the user is not allowed to do it. And we don't allow the user to scan for giftcards by looking at the response time.